app/services/teacher_training_public_api/sync_all_providers_and_courses.rb

```diff
@@ -1,15 +1,19 @@
 module TeacherTrainingPublicAPI
   class SyncAllProvidersAndCourses
-    def self.call(recruitment_cycle_year: ::RecruitmentCycle.current_year)
+    def self.call(recruitment_cycle_year: ::RecruitmentCycle.current_year, incremental_sync: true)
```

Comment:
How do we want to toggle the incremental_sync? Via the support interface? `support/settings/tasks`

Reply:
That's an interesting question. I definitely think you should be able to run if them the tasks. 

I wonder too if we should update the worker to occasionally run a full sync too. At the moment it runs every hour. We could potentially create a new attr like`fully_syned_at` and pass in `incramental_sync: false` if 3 days have passed and it's between 12pm & 7am or something like that.

Reply:
Yeah sounds like a good idea 👍 
Will have a think about the UI - sounds like we could simply replace the existing sync task button with one for full sync and another for partial sync. We may want to display a warning if full sync is activated during daytime?

Reply:
Apart from updating the provider sync task description we have agreed to leave the UI as is
https://ukgovernmentdfe.slack.com/archives/CN1MCQCHZ/p1620894860454000

Reply:
https://github.com/DFE-Digital/apply-for-teacher-training/pull/4712/commits/72eb011ee51ee4c615854c2eb547c650d3e1fb17

<img width="1181" alt="incrementally_sync_providers" src="https://user-images.githubusercontent.com/5256922/118110359-39450880-b3da-11eb-8ed8-6880d9cd2f79.png">

Reply:
https://github.com/DFE-Digital/apply-for-teacher-training/pull/4712/commits/c32c58c13890fadf4a876961b602c9d6961e0857

<img width="955" alt="incrementally_sync_providers" src="https://user-images.githubusercontent.com/5256922/118111001-04858100-b3db-11eb-9847-a18054c0ee5f.png">

Reply:
This is great 👌